kani-driver/src/main.rs

```diff
@@ -44,7 +43,9 @@ fn cargokani_main(input_args: Vec<OsString>) -> Result<()> {
 
     let mut goto_objs: Vec<PathBuf> = Vec::new();
     for symtab in &outputs.symtabs {
-        goto_objs.push(ctx.symbol_table_to_gotoc(symtab)?);
+        let goto_obj_filename = symtab.with_extension("out");
+        ctx.record_temporary_files(&[&goto_obj_filename]);
```

Comment:
Do we remove these files still? If so, wouldn't this break subsequent runs of `cargo kani`?

Reply:
They were being removed before, and I thought it best to keep that behaviour. @tedinski thoughts?

Reply:
The file removal behavior for `cargo kani` was something I went "huh that's strange" and just preserved what the original python scripts did. We seem to keep some things and delete others and I'm not sure why really.

I'm open to any behavior that makes sense here.

Reply:
@celinval? Vote?  Honestly, I think for cargo we should probably just keep everything since it goes into a target folder anyway.

Reply:
We do globs for what to link together in the end though, and that might motivate the deletion?

Reply:
For now, why don't we keep the current behaviour, and have an issue to have a principled set of things to keep/delete?

Reply:
https://github.com/model-checking/kani/issues/1736

Reply:
Testing showed that removing the files makes repeated cargo-kani fail